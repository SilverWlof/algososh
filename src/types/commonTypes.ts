export type TCircleStateType = {
    value: string | number;
    index?: number;
    color?: string;//TODO: determine colors from css or set fixed rgba strings
};