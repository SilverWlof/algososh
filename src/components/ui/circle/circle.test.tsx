import React from 'react';
import renderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

// 1 without letter
it('circle without letter', () => {
    const tree = renderer
        .create(<Circle />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//2 with letters
it('button with letters', () => {
    const tree = renderer
        .create(<Circle letter="test letters"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//3 with head
it('circle with head', () => {
    const tree = renderer
        .create(<Circle head="test text"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//4 with react in head
it('circle with react in head', () => {
    const tree = renderer
        .create(<Circle head={<Circle />} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

// with tail
it('circle with tail', () => {
    const tree = renderer
        .create(<Circle tail="test text" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//5 with react in tail
it('circle with react in tail', () => {
    const tree = renderer
        .create(<Circle tail={<Circle />} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//6 with index
it('circle with index', () => {
    const tree = renderer
        .create(<Circle index={0} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//7 with isSmall ===  true
it('circle with isSmall ===  true', () => {
    const tree = renderer
        .create(<Circle isSmall={true} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//8 in state equal :default
it('circle in state equal :default', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Default} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//9 in state equal :changing
it('circle in state equal :changing', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Changing} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//10 in state equal :modified
it('circle in state equal :modified', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Modified} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});