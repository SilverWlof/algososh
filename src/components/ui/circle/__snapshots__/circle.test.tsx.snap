// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button with letters 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    >
      test letters
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle in state equal :changing 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   changing"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle in state equal :default 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle in state equal :modified 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   modified"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle with head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="circle-head"
  >
    test text
  </div>
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle with index 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle with isSmall ===  true 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle  small default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle with react in head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="circle-head"
      />
      <div
        className="circle   default"
        data-cy="circle-border"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="circle-text"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="circle-tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;

exports[`circle with react in tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="circle-head"
      />
      <div
        className="circle   default"
        data-cy="circle-border"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="circle-text"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="circle-tail"
      />
    </div>
  </div>
</div>
`;

exports[`circle with tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="circle-tail"
  >
    test text
  </div>
</div>
`;

exports[`circle without letter 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circle-head"
  />
  <div
    className="circle   default"
    data-cy="circle-border"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circle-tail"
  />
</div>
`;
